raiden-ts/src/transfers/reducer.ts

```diff
@@ -33,27 +34,37 @@ function transferSecretReducer(
   action: transferSecret | transferSecretRegister.success,
 ): RaidenState {
   const secrethash = action.meta.secrethash;
-  // store when seeing unconfirmed, but registerBlock only after confirmation
+
+  const transferState = state[action.meta.direction][secrethash];
+  if (!transferState) return state;
+
+  // - Preserve the timestamp of the secret registration.
+  // - Preserve block number if a secret reveal message is received *after* the
+  //   secret is registered.
+  const previousRegisterBlock = transferState.secret?.registerBlock;
+  if (previousRegisterBlock) {
+    if (transferSecretRegister.success.is(action)) {
+      assert(action.payload.txBlock == previousRegisterBlock);
```

Comment:
Reducers never, ever throw. If some expectation isn't fulfilled, they should simply return the previous `state`.

Reply:
Also, we want to still update the `registerBlock` if it's different than the previous one. This shouldn't happen in practice, since we already check it's a `confirmed` action, but it's still valid from reducer's perspective.